test_programs/execution_success/struct_equality/src/main.nr

```diff
@@ -0,0 +1,19 @@
+// This test demonstrates how SSA optimizations breaks down a struct equality assertion
+// to assert directly on the struct's fields rather than using a predicate.
+
+struct MyStruct {
+    foo: Field,
+    bar: bool,
+    baz: [u8; 3]
+}
+
+impl Eq for MyStruct {
+    fn eq(self, other: MyStruct) -> bool {
+        (self.foo == other.foo) & (self.bar == other.bar) & (self.baz == other.baz)
+    }
+}
+
+fn main(x: MyStruct) {
+    let y = MyStruct { foo: 42, bar: true, baz: [1, 2, 3] };
+    assert(x.eq(y));
+}
```

Comment:
Is this possible to promote to a SSA unit test? It seems cumbersome to have it as an execution_success test when we want to test if an optimization occurs and not just that it ran correctly.

Reply:
Yep, I can change this to a test within rust. Will do this tomorrow.